src/router/index.ts

```diff
@@ -0,0 +1,422 @@
+    },
+  };
+}
+
+// Replaces history on any user interaction with forward and backwards buttons
+// with the current window.history.state
+window.addEventListener("popstate", e => {
+  if (e.state) replace(e.state);
+});
+
+export function navigate(
+  route: Route | string,
+  opts?: { replace?: boolean },
+): void {
+  if (typeof route === "string") {
+    route = pathToRoute(route);
+  }
+
+  // If we don't explicitly pass a anchor tag hash or query string, we should reset it globally
+  if (route.type === "projects" && !route.params.hash) {
+    route.params = { ...route.params, hash: null, search: null };
+  }
+
+  // If activeRoute and newRoute are type === "projects" combine the route params
+  const activeRoute = get(activeRouteStore);
+  if (route.type === "projects" && activeRoute.type === "projects") {
+    route.params = { ...activeRoute.params, ...route.params };
+  }
+
+  if (opts?.replace) {
+    replace(route);
+  } else {
+    push(route);
+  }
+}
+
+export const push = (newRoute: Route): void => {
+  const history = get(historyStore);
+
+  // Limit history to a maximum of 10 steps. We shouldn't be doing more than
+  // one subsequent pop() anyway.
+  historyStore.set([...history, newRoute].slice(-10));
+  activeRouteStore.set(newRoute);
+  window.history.pushState(newRoute, documentTitle, routeToPath(newRoute));
+};
+
+export const pop = (): void => {
+  const history = get(historyStore);
+  const newRoute = history.pop();
+  if (newRoute) {
+    historyStore.set(history);
+    activeRouteStore.set(newRoute);
+    window.history.back();
+  }
+};
+
+function replace(newRoute: Route): void {
+  historyStore.set([newRoute]);
+  activeRouteStore.set(newRoute);
+  window.history.replaceState(newRoute, documentTitle, routeToPath(newRoute));
+}
+
+export const initialize = () => {
+  const { pathname, search, hash } = window.location;
+  replace(pathToRoute(pathname + search + hash));
+};
+
+export function pathToRoute(path: string | null): Route {
+  if (!path) {
+    return { type: "404" };
+  }
+
+  const url = new URL(path, "https://app.radicle.xyz");
+  // Pathname starts usually with a "/", we remove it to avoid bad interpretations
+  const segments = url.pathname.substring(1).split("/");
+
+  const type = segments.shift();
+  switch (type) {
+    case "registration": {
+      const nameOrDomain = segments.shift();
+      const activeView = segments.shift();
+      const owner = url.searchParams.get("owner");
+      const retry = url.searchParams.get("retry");
+
+      if (nameOrDomain) {
+        if (
+          activeView === "checkNameAvailability" ||
+          activeView === "register"
+        ) {
+          return {
+            type: "registration",
+            params: {
+              activeView: {
+                type: activeView,
+                params: { nameOrDomain, owner },
+              },
+            },
+          };
+        } else if (activeView === undefined) {
```

Comment:
```suggestion
        } else if (!activeView) {
```

Reply:
Done in 5f3963bd91ebbc552e216dd713136dafa383d1e6